src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/IHtmlHelper.cs

```diff
@@ -169,7 +169,12 @@ public interface IHtmlHelper
 
         /// <summary>
         /// Returns an &lt;input&gt; element of type "checkbox" with value "true" and an &lt;input&gt; element of type
-        /// "hidden" with value "false".
+        /// "hidden" with value "false" for the specified <paramref name="expression"/>. Adds a "checked" attribute to
+        /// the "checkbox" element if first non-<c>null</c> value found is <c>true</c>. Looks for
```

Comment:
I did the changes out-of-order and got to `CheckBox()` and `RadioButton()` near the end. I was in the habit of describing helpers that find the first value and use it in `value` attributes. Agree applying that pattern here might be less readable.

If we stick w/ the pattern, "if *the* first" may make it more clear that we aren't talking about the checkbox element being first 😺 

Let's chat tomorrow as you suggested.